clients/js/src/embeddings/OpenAIEmbeddingFunction.ts

```diff
@@ -95,63 +89,50 @@ export class OpenAIEmbeddingFunction implements IEmbeddingFunction {
     this.model = openai_model || "text-embedding-ada-002";
   }
 
-  private async loadClient() {
-    // cache the client
-    if (this.openaiApi) return;
+  public async generate(texts: string[]): Promise<number[][]> {
+    const openaiApi = await this.getOpenAIClient();
 
+    return await openaiApi
+      .createEmbedding({
+        model: this.model,
+        input: texts,
+      })
+      .catch((error: any) => {
```

Comment:
Nit: Looks like you might be able to get rid of this catch since you're just forwarding it along to the caller?

Reply:
You are right. We don't have a reason to rethrow the exception unless we have an EF specific error wrapper 🤔